pallets/gear/src/journal_builder.rs

```diff
@@ -0,0 +1,164 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Config, ExecutionEnvironment, ExtManager, GasAllowanceOf, GasBalanceOf, Pallet};
+use alloc::vec::Vec;
+use common::{storage::Limiter, CodeStorage, Origin};
+use core_processor::{
+    common::{ExecutableActorData, JournalNote, PrechargedDispatch},
+    configs::BlockConfig,
+    ContextChargedForInstrumentation,
+};
+use frame_support::traits::{Get, Randomness};
+use gear_core::{ids::ProgramId, message::StoredDispatch};
+use parity_scale_codec::Encode;
+use sp_runtime::traits::UniqueSaturatedInto;
+
+pub(crate) struct JournalBuilder<'a, T: Config, F> {
```

Comment:
Completelly disagree wiith naming. This one is just another layer around smart-contract execution. So, name must be something like prepare_and_proccess_message_execution().

Reply:
Layer that builds journal list as the result

Reply:
Is not an argument